controllers/amazoncloudwatchagent_controller.go

```diff
@@ -40,6 +43,80 @@ type Params struct {
 	Config   config.Config
 }
 
+func (r *AmazonCloudWatchAgentReconciler) findCloudWatchAgentOwnedObjects(ctx context.Context, owner v1alpha1.AmazonCloudWatchAgent) (map[types.UID]client.Object, error) {
+	// Define a map to store the owned objects
+	ownedObjects := make(map[types.UID]client.Object)
+	selector := manifestutils.SelectorLabels(owner.ObjectMeta, "*")
+	delete(selector, "app.kubernetes.io/component") //ignore components
+	listOps := &client.ListOptions{
+		Namespace:     owner.Namespace,
+		LabelSelector: labels.SelectorFromSet(selector),
+	}
+	// Define lists for different Kubernetes resources
+	configMapList := &corev1.ConfigMapList{}
+	serviceList := &corev1.ServiceList{}
+	serviceAccountList := &corev1.ServiceAccountList{}
+	deploymentList := &appsv1.DeploymentList{}
+	statefulSetList := &appsv1.StatefulSetList{}
+	daemonSetList := &appsv1.DaemonSetList{}
+	var err error
+
+	// List ConfigMaps
+	err = r.List(ctx, configMapList, listOps)
+	if err != nil {
+		return nil, err
```

Comment:
Can you specify the resource it failed to list?
```suggestion
		return nil, fmt.Errorf("failed to list ConfigMaps: %w", err)
```

Reply:
It already does do that : 
![Screenshot 2024-10-22 at 2 47 37 PM](https://github.com/user-attachments/assets/93d1fa16-618f-4065-894e-746d0c3948dc)